### PR TITLE
Excluding output image formats .hddimage, .iso, and .tar.bz2 from dep…

### PR DESCRIPTION
**Pull request by SKumarDas**
Deprecating ..hddimage, .iso, and .tar.bz2 image formats as we don't need it. 
Please help to review and merge it.

Regards,
Sapan Das

### CODE REVIEW

File: conf/variant/common/local.conf [hunk 1]
```diff
@@ -20,3 +20,7 @@ DIRFILES_pn-nativesdk-cmake = "1"
 
 # Don't install any extra devel packages just because we're running on QEMU
 POKYQEMUDEPS_remove = "packagegroup-core-device-devel"
+
+# Don't build .hddimg and .iso image formats as we don't need it.
```
Comment (GordanM) [inline conf/variant/common/local.conf]: No need for the `.` at the end of the line here. And s/it/them

Comment (tobsan) [reply conf/variant/common/local.conf]: We are using `.rpi-sdimg` and `.wic` for pi and NUC, respectively.

File: classes/core-image-pelux.bbclass [hunk 2]
```diff
@@ -40,3 +40,4 @@ toolchain_create_sdk_env_script_append() {
 IMAGE_ROOTFS_SIZE ?= "1000000"
 IMAGE_FSTYPES ?= "ext3 sdcard"
 IMAGE_FSTYPES += "ext3.gz"
```
Comment (tobsan) [inline classes/core-image-pelux.bbclass]: Do we need this filetype at all?

**Comment by breakreturn** [comment]
Since we are in some way providing experimental support for TX-2, or at least have the ambition (the last time I checked) to do so, wouldn't we take a step away from that with this? While also breaking things for current users of TX-2?

Qemu isn't affected by the change either I guess?

**Comment by GordanM** [comment]
@breakreturn good point! This should be taken into account. Could you please connect with someone from their team @SKumarDas ?

File: conf/variant/common/local.conf [hunk 3]
```diff
@@ -20,3 +20,7 @@ DIRFILES_pn-nativesdk-cmake = "1"
...
+NOHDD = "1"
+NOISO = "1"
```
Comment (erikboto) [inline conf/variant/common/local.conf]: Can we use the more override-friendly version ?= to assign NOHDD and NOISO here?

File: classes/core-image-pelux.bbclass [hunk 4]
```diff
@@ -40,3 +40,4 @@ toolchain_create_sdk_env_script_append() {
...
+IMAGE_FSTYPES_remove = "tar.bz2"
```
Comment (erikboto) [inline classes/core-image-pelux.bbclass]: Is the assignment on line 41 not happening? Seems weird that we would need this _remove line if the assignment is happening. And if it's not happening, it should probably be removed for clarity or changed to a non-conditional assignment to force it.

Also line 42 seems weird if we want ext3.gz it should just be added in the line 41.

Comment (tobsan) [reply classes/core-image-pelux.bbclass]: Isn't the assignment on 41 a strong default, so it may be overridden by, for example, machine configurations? I agree that we should make this clearer though.

Comment (erikboto) [reply classes/core-image-pelux.bbclass]: Yes it can be overridden, which is good. But why are we then explicitly adding ext3.gz and removing tar.bz2 afterwards? What if someone wants tar.bz2? Shouldn't we either set our preferred defaults using ?=, and then not touch if if someone chooses to override it?

File: conf/variant/common/local.conf (hunk 3 above)
Comment (erikboto) [reply conf/variant/common/local.conf]: I guess it can still be overridden in a machine specific local.conf if it's done after including common/local.conf, so I guess it's ok to keep the assigment as it is now.

File: classes/core-image-pelux.bbclass (hunk 2 above)
Comment (FlorentRevest) [reply classes/core-image-pelux.bbclass]: I need the ext3.gz for SWUpdate artifacts

Comment (SKumarDas) [reply classes/core-image-pelux.bbclass]: @FlorentRevest, is it is okay to remove .tar.bz2? Or you will need this image format as well?

Comment (FlorentRevest) [reply classes/core-image-pelux.bbclass]: @SKumarDas you can remove it, I don't need that one

Comment (SKumarDas) [reply classes/core-image-pelux.bbclass]: @breakreturn Could you please conform what image format you use in qemo? Do you use tar.bz2 in any case?

Comment (breakreturn) [reply classes/core-image-pelux.bbclass]: @SKumarDas Maybe @jonte or @erikboto can support with this.

File: conf/variant/common/local.conf [hunk 5]
```diff
@@ -20,3 +20,10 @@ DIRFILES_pn-nativesdk-cmake = "1"
...
+
+# Removing tar.bz2 image format as it is not used. Whoever needs it, they can add it in respective variant's local.conf file
```
Comment (tobsan) [inline conf/variant/common/local.conf]: Perhaps we can reword this as: whoever needs this will supply it in their own ``local.conf``. The variant concept is something relatively local to us.
